lib/src/browser/disposable_browser.dart

```diff
@@ -203,17 +204,17 @@ class Disposable implements disposable_common.Disposable {
     _disposable
       ..onDisposeHandler = onDispose
       ..onWillDisposeHandler = onWillDispose;
-    return _disposable.dispose().then((_) {
-      // We want the description to be the runtime type of this
-      // object, not the proxy disposable, so we need to reset
-      // the leak flag here.
-      flagLeak(runtimeType.toString());
-    });
+    // We want the description to be the runtime type of this
+    // object, not the proxy disposable, so we need to set
+    // the leak flag here, before we delegate the `dispose`
+    // call.
+    flagLeak();
+    return _disposable.dispose();
   }
 
   @override
   void flagLeak([String description]) {
-    _disposable.flagLeak(description);
+    _disposable.flagLeak(description ?? runtimeType.toString());
```

Comment:
IIRC `runtimeType` is very slow (and also isn't useful in minified builds). We should be extremely cautious about where we use it.

Reply:
Oh, looks like we were already using it 😦

Reply:
It's used in a few places in our platform pieces. Maybe we should audit that.